### PR TITLE
OCPBUGS-31510: change the analysis to not include ImageStreamTag

### PR DESCRIPTION
**Pull request by prb112**
#### Description:

ImageStreamTag is improperly considered with the rule.yaml

#### Rationale:

https://issues.redhat.com/browse/OCPBUGS-31510

#### Review Hints:

The logic changes are for rhcos4-disa-stig.

### CODE REVIEW

**Comment by prb112** [comment]
Checked a prior run with PR 11593 - found the same failures for `e2e-stig-imagestream-sets-schedule` I think it might be expected, and other elements are requiring a retest.

File: applications/openshift/registry/imagestream_sets_schedule/rule.yml [hunk 1]
```diff
@@ -42,22 +42,20 @@ ocil: |-
   <pre>oc get imagestream  --all-namespaces -o jsonpath='{{{ jqfilter }}}'</pre>
   Alternatively, to view a list of ImageStreams that do not schedule updates,
   run:
-  <pre>oc get imagestreams -A -ojson | jq '.items[] | select(.spec.tags[]?.importPolicy.scheduled != true) | .metadata.name' | sort | uniq</pre>
+  <pre>oc get imagestreams -A -ojson | jq -r '.items[] | select(.spec.tags[]?.importPolicy.scheduled != true) | "\(.metadata.namespace),\(.metadata.name)"' | sort | uniq</pre>
```
Comment (Vincent056) [inline applications/openshift/registry/imagestream_sets_schedule/rule.yml:45]: Wondering if we also need `select(.from.kind != "ImageStreamTag")` here

Comment (prb112) [reply applications/openshift/registry/imagestream_sets_schedule/rule.yml:45]: Sure - I can add that. will take care of it today

**Comment by prb112** [comment]
Hey @Vincent056 - I've updated the command within the `<pre>` text. Thank you, Paul

**Comment by prb112** [comment]
Hi @xiaojiey have you had any updates on testing from your end? Thanks! Paul

File: applications/openshift/registry/imagestream_sets_schedule/rule.yml [hunk 2]
```diff
@@ -42,22 +42,20 @@ ocil: |-
   <pre>oc get imagestream  --all-namespaces -o jsonpath='{{{ jqfilter }}}'</pre>
```
[the remark this answers is not in the document]
Comment (prb112) [reply applications/openshift/registry/imagestream_sets_schedule/rule.yml:42]: Hi @xiaojiey I think I follow you. That seems reasonable - however, it'll have to be hard coded. jsonpath does not support the filter above.

[the remark this answers is not in the document]
Comment (prb112) [reply applications/openshift/registry/imagestream_sets_schedule/rule.yml]: This should be resolved @xiaojiey - Thank you for the diligence

**Comment by xiaojiey** [comment]
verification pass with 4.16.0-0.nightly-2024-05-07-025557 + ghcr.io/complianceascode/k8scontent:11783. Details seen from the comment of the bug https://issues.redhat.com/browse/OCPBUGS-31510

**Comment by xiaojiey** [comment]
/

**Comment by prb112** [comment]
Hey @Vincent056 or @rhmdnd do you mind reviewing the PR? @xiaojiey has kindly tested. Thanks, Paul

**Review by yuumasato — APPROVED** [review]
Thank you for the fix.

**Comment by yuumasato** [comment]
Skipping the Ansible hardening tests.

**Pull request merged**